app/models/git/repository.rb

```diff
@@ -134,6 +134,7 @@ def rugged_repo
         cmd = [
           "git clone",
           "--bare",
+          "-c core.sharedRepository=true",
```

Comment:
This tells git to honour group not user. It's set in the `.git/config` of each repo. We need to manually update old ones but this should fix it for new ones.